tools/introspection.c

```diff
@@ -512,6 +512,11 @@ set_id_buffer(struct ulp_process *process, unsigned char *patch_id)
 
   DEBUG("advertising live patch ID to libpulp.");
 
+  if (!process->all_threads_hijacked) {
+    WARN("not all threads hijacked.");
+    return 1;
+  }
+
```

Comment:
To be fully accurate, this function, `set_id_buffer`, doesn't actually require that all threads have been previously attached to, because it doesn't execute any routines from libpulp.so. Functions from libpulp.so, such as `__ulp_trigger`, actually require it, because they access shared variables, such as the locks in the malloc and dlopen implementations.

Nonetheless, `set_id_buffer` does require that _at least_ the main thread is attached to, because it ultimately uses `PTRACE_PEEKDATA`, and testing for the hijacking of all threads works fine.

So, this change looks good to me, but it would be nice if this explanation was recorded in the commit message.

Reply:
May have been somewhat naive of me, but I figured that every critical section function warranted this check. Will revise commit message with explanation.